File: day04/sol.c
#include <stdio.h>

int main() {
  printf("Day 4: C\n");
  int p1 = 0, p2 = 0;
  int a0 = 0, a1 = 0, b0 = 0, b1 = 0;
  while (scanf("%d-%d,%d-%d", &a0, &a1, &b0, &b1) > 0) {
    if (a0 <= b0 && a1 >= b1 || b0 <= a0 && b1 >= a1) p1++;
    if (a0 <= b1 && b0 <= a1) p2++;
  }
  printf("Part 1: %20d\n", p1);
  printf("Part 2: %20d\n", p2);
}

File: day06/sol.ml
let solve nchars input =
  let rec go c xs =
    if List.length (List.sort_uniq compare (List.init nchars (String.get xs))) == nchars
    then c
    else go (c+1) (String.sub xs 1 (String.length xs - 1)) in
  go nchars input

let () =
  print_endline "Day 6: OCaml";
  let input = read_line () in
  Printf.printf "Part 1: %20d\n" (solve 4 input);
  Printf.printf "Part 2: %20d\n" (solve 14 input);
